src/models/feedDao.js

```diff
@@ -0,0 +1,227 @@
+            SELECT
+                feeds.id AS id,
+                u.nickname AS userNickname,
+                b.image_url AS badge,
+                imgurl,
+                feeds.content,
+                CASE
+                    WHEN comment IS NOT NULL THEN comment
+                    ELSE 0
+                END AS comment,
+                is_challenge AS challenge, 
+                DATE_FORMAT(feeds.created_at, '%Y-%m-%d') AS created_at,
+                CASE
+                    WHEN feeds.user_id = ? THEN true
+                    ELSE false
+                END AS isMyPost
+            FROM 
+                feeds 
+            LEFT JOIN 
+                users u ON feeds.user_id = u.id
+            LEFT JOIN 
+                badges b ON u.badge_id = b.id
+            LEFT JOIN (
+                SELECT
+                    feed_id,
+                    COUNT(id) AS comment
+                FROM
+                    comments
+                GROUP BY feed_id
+                    ) AS comments ON feeds.id = comments.feed_id
+            LEFT JOIN (
+                SELECT
+                    feed_id,
+                JSON_ARRAYAGG(
+                    JSON_OBJECT(
+                        "id" , id,
+                        "url" , url
+                    )
+                ) AS imgurl
+                FROM
+                    feed_images
+                GROUP BY feed_id
+            ) AS feed_images ON feeds.id = feed_images.feed_id
+            ORDER BY
+                feeds.created_at DESC
+            LIMIT ? OFFSET ?
+            `,
+            [userId, limit, offset]
+        );
+        return Feeds;
+    }
+
+    // 총 피드 갯수
+    const getFeedCount = async () => {
+        const feedCount = await AppDataSource.query(`
+            SELECT COUNT(*) 
+            AS total 
+            FROM 
+                feeds
+        `);
+        return feedCount[0].total;
+      }
+
+    // 피드 작성
+    const addFeeds = async(userId, content, challenge, imageUrl) => {
+        // 중복 이미지
+        imageUrl = Array.from(new Set(imageUrl)).slice(0, 3);
+        // challenge 체크박스 값
+        const isChallenge = challenge === '0';
+
+        await AppDataSource.transaction(async (transactionManager) => {
+            const{ insertId:feedId } = await transactionManager.query(`
+                INSERT INTO feeds (
+                    user_id,
+                    content,
+                    is_challenge
+                ) VALUES (?, ?, ?)`,
+                [userId, content, isChallenge]
+            );
+            for (let i = 0; i < imageUrl.length; i++){
+                await transactionManager.query(`
+                    INSERT INTO feed_images (
+                        url,
+                        feed_id
+                    ) VALUES (?, ?)`,
+                    [imageUrl[i], feedId]
+                );
+            }   
+        })
+    }
+
+    // 피드 전체 삭제
+    const deleteFeeds = async(feedId, userId, challenge) => {
```

Comment:
```suggestion
    const deleteFeeds = async(feedId) => {
```
- feedId 자체가 primary key로 유일한 고유값이기 때문에, 지울 때는 키 하나만 있어도 됩니다 다른 변수는 비교할 필요 없을 것 같습니다 :)

Reply:
네! 지우고 수정했습다